crates/e2e/tests/e2e/services.rs

```diff
@@ -60,10 +60,39 @@ macro_rules! tx {
     };
 }
 
+#[macro_export]
+macro_rules! tx_safe {
+    ($acc:ident, $safe:ident, $call:expr) => {{
+        let call = $call;
+        $crate::tx!(
+            $acc,
+            $safe.exec_transaction(
+                call.tx.to.unwrap(),
+                call.tx.value.unwrap_or_default(),
+                ::ethcontract::Bytes(call.tx.data.unwrap_or_default().0),
+                Default::default(),
+                Default::default(),
+                Default::default(),
+                Default::default(),
+                Default::default(),
+                Default::default(),
+                $crate::services::safe_validator_signature($acc.address()),
+            )
+        );
+    }};
+}
+
 pub fn to_wei(base: u32) -> U256 {
     U256::from(base) * U256::from(10).pow(18.into())
 }
 
+pub fn safe_validator_signature(owner: H160) -> Bytes<Vec<u8>> {
+    let mut signature = vec![0; 65];
+    signature[12..32].copy_from_slice(owner.as_bytes());
+    signature[64] = 1;
```

Comment:
I don't know if the signature is initialized correctly.
Is there maybe some smart contract code you can link here which could explain that?

Reply:
Added comment explaining what signature scheme we are using with links to the Safe docs and SC code that verifies it.